### PR TITLE
(PC-11727) feat(IdentityCheck): add redirection to Ubble webview

### PR DESCRIPTION
**Pull request by antoinewg**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-11727

## Checklist

I have:

- [x] Made sure the title of my PR follows the [convention](../doc/standards/pr-title.md) `($jira) $type($scope): $summary`.
- [ ] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** or **github dev name** for any added TODO / FIXME.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" : `https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a`
- [x] Made sure translations file is up to date (`yarn translations:extract`)

## Screenshots

| iOS | Android | Mobile - Chrome | Desktop - Chrome | Desktop - Safari |
| --: | ------: | --------------: | ---------------: | ---------------: |
|     |         |                 |                  |                  |

### CODE REVIEW

File: src/features/identityCheck/api.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { useMutation } from 'react-query'
+
+import { api } from 'api/api'
+import { IdentificationSessionResponse } from 'api/gen'
+
+interface RequestIdentificationUrlProps {
+  onSuccess: (data: IdentificationSessionResponse) => void
+  onError: () => void
+  onSettled: () => void
+}
```
Comment (BenStirrup) [inline src/features/identityCheck/api.ts:10]:
```suggestion
type Props = UseMutationOptions<IdentificationSessionResponse>
```

File: src/features/identityCheck/components/SomeAdviceBeforeIdentityCheckModal.tsx [hunk 2]
```diff
@@ -10,40 +13,67 @@ import { IdCard } from 'ui/svg/icons/IdCard'
...
+import { useNavigation } from '@react-navigation/native'
+import { UseNavigationType } from 'features/navigation/RootNavigator'
 
 interface Props {
   visible: boolean
   hideModal: () => void
 }
 
+const redirectUrl = `https://${env.WEBAPP_V2_DOMAIN}/verification-identite/fin`
+const identificationUrl = 'https://id.ubble.ai/70f01f19-6ec5-4b14-9b30-2c493e49df15'
```
Comment (kopax-polyconseil) [inline src/features/identityCheck/components/SomeAdviceBeforeIdentityCheckModal.tsx:25]: Ceci serait mieux dans les dotenv (`.env`)

File: src/features/identityCheck/components/SomeAdviceBeforeIdentityCheckModal.tsx [hunk 3]
```diff
@@ -10,40 +13,67 @@ import { IdCard } from 'ui/svg/icons/IdCard'
...
-      <Instruction title={t`Cadre l’intégralité de ton document`} Icon={IdCard} />
-    </Instructions>
-    <ButtonPrimary title={t`J'ai compris`} onPress={hideModal} />
-  </AppModal>
-)
+}) => {
+  const { navigate } = useNavigation<UseNavigationType>()
+
+  const { mutate: requestIdentificationUrl } = useRequestIdentificationUrl({
+    onSuccess: (data: IdentificationSessionResponse) => {
```
Comment (kopax-polyconseil) [inline src/features/identityCheck/components/SomeAdviceBeforeIdentityCheckModal.tsx:34]: `onSuccess(`

File: src/features/identityCheck/components/SomeAdviceBeforeIdentityCheckModal.tsx [hunk 4]
```diff
@@ -10,40 +13,67 @@ import { IdCard } from 'ui/svg/icons/IdCard'
...
+      navigate('IdentityCheckWebview', { identificationUrl })
+    },
+    onError: () => {
```
Comment (kopax-polyconseil) [inline src/features/identityCheck/components/SomeAdviceBeforeIdentityCheckModal.tsx:37]: `onError(`

File: src/features/identityCheck/components/SomeAdviceBeforeIdentityCheckModal.tsx (hunk 2 above)
Comment (antoinewg) [reply src/features/identityCheck/components/SomeAdviceBeforeIdentityCheckModal.tsx:25]: c'est temporaire (on recevra cet url du backend au call précédent :) je mets un TODO

File: src/features/identityCheck/components/SomeAdviceBeforeIdentityCheckModal.tsx [hunk 5]
```diff
@@ -10,40 +13,67 @@ import { IdCard } from 'ui/svg/icons/IdCard'
...
 import { ColorsEnum, getSpacing, Typo } from 'ui/theme'
+import { useNavigation } from '@react-navigation/native'
+import { UseNavigationType } from 'features/navigation/RootNavigator'
 
 interface Props {
   visible: boolean
   hideModal: () => void
 }
 
+const redirectUrl = `https://${env.WEBAPP_V2_DOMAIN}/verification-identite/fin`
```
Comment (BenStirrup) [inline src/features/identityCheck/components/SomeAdviceBeforeIdentityCheckModal.tsx:24]: On a une variable `WEBAPP_V2_URL` :
```suggestion
const redirectUrl = `${env.WEBAPP_V2_URL}/verification-identite/fin`
```

Comment (kopax-polyconseil) [reply src/features/identityCheck/components/SomeAdviceBeforeIdentityCheckModal.tsx:24]: Elle n'est pas dans `env`, mais dans `import { WEBAPP_V2_URL } from 'libs/environment/useWebAppUrl'`

**Review by BenStirrup — APPROVED** [review]

**Review by Lucasbeneston — APPROVED** [review]

**Pull request merged**